pkg/xen-tools/xen-info

```diff
@@ -15,22 +15,40 @@ waited=0
 ID=$(xl domid "$1" 2>/dev/null)
 while [ -z "$ID" ] && [ "$waited" -lt 60 ]; do
         ID=$(xl domid "$1" 2>/dev/null)
-        sleep 3
-        waited=$((waited + 3))
+        if [ -z "$ID" ]; then
+          sleep 3
+          waited=$((waited + 3))
+        fi
 done
 [ -z "$ID" ] && bail "Couldn't find domain ID for domain $1"
 
 # we expect to get rbpscd where every letter can also be a dash (-)
 # Name    ID    Mem    VCPUs    State    Time(s)
-case $(xl list "$ID" | awk '{st=$5;} END {print st;}') in
-   *d) STATUS=halting ;;
-  *c*) STATUS=broken  ;;
-  *s*) STATUS=halting ;;
-  *p*) STATUS=paused  ;;
-  *b*) STATUS=running ;;
-   r*) STATUS=running ;;
-    *) STATUS=$(cat /dev/status) ;;
-esac
+waited=0
+STATUS=
+while [ -z "$STATUS" ] && [ "$waited" -lt 60 ]; do
+  case $(xl list "$ID" | awk '{st=$5;} END {print st;}') in
+     *d) STATUS=halting ;;
+    *c*) STATUS=broken  ;;
+    *s*) STATUS=halting ;;
+    *p*) STATUS=paused  ;;
+    *b*) STATUS=running ;;
+     r*) STATUS=running ;;
+      *) STATUS=$(cat /dev/status) ;;
+  esac
+  if [ -z "$STATUS" ]; then
+    sleep 3
+    waited=$((waited + 3))
+    ID=$(xl domid "$1" 2>/dev/null)
+    [ -z "$ID" ] && bail "Couldn't find domain ID for domain $1"
+  fi
+done
+
+if [ -z "$STATUS" ]; then
+  ID=$(xl domid "$1" 2>/dev/null)
+  [ -z "$ID" ] && bail "Couldn't find domain ID for domain $1"
+  bail "Couldn't determine state of domain $1"
+fi
```

Comment:
You already checked the ID in the last iteration of the loop above so why do you need to check again?

Reply:
Added that to double-check if the domain is still running before xen-info returns undermined status error.
Removed it.